sdlf-ddk-lightweight/data_lake/stages/foundations_stage.py

```diff
@@ -0,0 +1,538 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+from pathlib import Path
+from typing import Any, Dict
+
+import aws_cdk as cdk
+from aws_ddk_core.pipelines.stage import DataStage
+import aws_cdk.aws_dynamodb as DDB
+from aws_cdk.aws_iam import Effect, PolicyDocument, PolicyStatement, Role, ServicePrincipal, ManagedPolicy
+from aws_cdk.aws_kms import Key, IKey
+from aws_cdk.aws_lakeformation import CfnResource
+from aws_cdk.aws_lambda import Code, Function, Runtime, LayerVersion, IFunction, ILayerVersion
+from aws_cdk.aws_s3 import Bucket, BucketEncryption, BlockPublicAccess, BucketAccessControl, IBucket
+from aws_cdk.aws_ssm import StringParameter
+from aws_cdk.custom_resources import Provider
+from aws_cdk.aws_s3_deployment import BucketDeployment, ServerSideEncryption, Source
+from aws_ddk_core.resources import S3Factory, KMSFactory, LambdaFactory
+
+class FoundationsStage(DataStage):
```

Comment:
Could you explain what is the reason for making `FoundationsStage` a `DataStage`? It does not seem to produce an event or be chained with any other data transformation or ingestion stage.

Could this be a stack inheriting from `BaseStack` instead?